internal/controller/controllers/clusterdeployments_controller_test.go

```diff
@@ -244,6 +244,23 @@ var _ = Describe("cluster reconcile", func() {
 
 				validateCreation(cluster)
 			})
+
+			It("pull secret in string data", func() {
```

Comment:
I'm not sure that data and string data are working together, this is the model that hive is using https://github.com/openshift/hive/blob/master/docs/using-hive.md#pull-secret and the same will be used by the users.